include/nbl/builtin/hlsl/shapes/beziers.hlsl

```diff
@@ -40,89 +40,204 @@ namespace shapes
         numRealValues = 2;
         return float2(sqrtD - b, sqrtD + b);
     }
-
+    
+    template<typename float_t>
     struct QuadraticBezier
     {
-        float2 A;
-        float2 B;
-        float2 C;
+        using float2_t = vector<float_t, 2>;
+        using float3_t = vector<float_t, 3>;
 
-        static QuadraticBezier construct(float2 a, float2 b, float2 c)
+        struct ArcLengthPrecomputedValues
```

Comment:
we can merge this struct and `QuadBezierAnalyticArcLengthCalculator` together and do the precomputation in the constructor.

but let's not do that right now cause we'll have to include `beziers.hlsl` inside c++ and fix incompatibilities and shit.

Reply:
irrevelant now